src/server/pachyderm_test.go

```diff
@@ -2164,82 +2166,131 @@ func TestStopPipeline(t *testing.T) {
 	require.Equal(t, "foo\n", buffer.String())
 }
 
-func TestPipelineAutoScaledown(t *testing.T) {
+func TestStandby(t *testing.T) {
 	if testing.Short() {
 		t.Skip("Skipping integration tests in short mode")
 	}
 
 	c := getPachClient(t)
-	defer require.NoError(t, c.DeleteAll())
-	// create repos
-	dataRepo := tu.UniqueString("TestPipelineAutoScaleDown")
-	require.NoError(t, c.CreateRepo(dataRepo))
-	// create pipeline
-	pipelineName := tu.UniqueString("pipeline-auto-scaledown")
-	parallelism := 4
-	scaleDownThreshold := time.Duration(10 * time.Second)
-	_, err := c.PpsAPIClient.CreatePipeline(
-		context.Background(),
-		&pps.CreatePipelineRequest{
-			Pipeline: client.NewPipeline(pipelineName),
-			Transform: &pps.Transform{
-				Cmd: []string{"sh"},
-				Stdin: []string{
-					"echo success",
+	t.Run("ChainOf10", func(t *testing.T) {
+		require.NoError(t, c.DeleteAll())
+
+		dataRepo := tu.UniqueString("TestStandby_data")
+		require.NoError(t, c.CreateRepo(dataRepo))
+
+		numPipelines := 10
+		pipelines := make([]string, numPipelines)
+		for i := 0; i < numPipelines; i++ {
+			pipelines[i] = tu.UniqueString("TestStandby")
+			input := dataRepo
+			if i > 0 {
+				input = pipelines[i-1]
+			}
+			_, err := c.PpsAPIClient.CreatePipeline(context.Background(),
+				&pps.CreatePipelineRequest{
+					Pipeline: client.NewPipeline(pipelines[i]),
+					Transform: &pps.Transform{
+						Cmd: []string{"true"},
+					},
+					Input:   client.NewAtomInput(input, "/*"),
+					Standby: true,
 				},
-			},
-			ParallelismSpec: &pps.ParallelismSpec{
-				Constant: uint64(parallelism),
-			},
-			ResourceRequests: &pps.ResourceSpec{
-				Memory: "100M",
-			},
-			Input:              client.NewAtomInput(dataRepo, "/"),
-			ScaleDownThreshold: types.DurationProto(scaleDownThreshold),
+			)
+			require.NoError(t, err)
+		}
+
+		require.NoErrorWithinT(t, time.Second*30, func() error {
```

Comment:
Is there a reason why you are not using NoErrorWithinTRetry here?

Reply:
Good call, looks like I should be.